pkg/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -0,0 +1,180 @@
+/*
+   Copyright The containerd Authors.
```

Comment:
Please add the link to the original source

Reply:
Also please add code comment to explain why the nerdctl package can't be just imported here